### ISSUE TITLE
llvm-nm -opaque-pointers option missing

### ISSUE THREAD

**Issue by Brandon-T** [llvm/llvm-project#55506]
When listing symbols in `rust` for `aarch64` binaries on MacOS, I do:
```shell
/src/third_party/llvm-build/Release+Asserts/bin/llvm-nm /src/out/ios_Debug_arm64/custom_rust/out/arm64/aarch64-apple-ios/debug/libcustom_rust.a --extern-only --no-sort --just-symbol-name --opaque-pointers | grep error:
```
which results in: `error: : unknown argument '--opaque-pointers'`

If I remove the `--opaque-pointers` option, then the result is:
```console
error: /src/out/ios_Debug_arm64/custom_rust/out/arm64/aarch64-apple-ios/debug/libcustom_rust.a(hide.o): Opaque pointers are only supported in -opaque-pointers mode (Producer: 'LLVM15.0.0git' Reader: 'LLVM 15.0.0git')

/src/out/ios_Debug_arm64/custom_rust/out/arm64/aarch64-apple-ios/debug/libcustom_rust.a:addtf3.o: no symbols
/src/out/ios_Debug_arm64/custom_rust/out/arm64/aarch64-apple-ios/debug/libcustom_rust.a:comparetf2.o: no symbols
```

Is there a way to get around this issue on `llvm-nm`?:
```console
llvm-nm, compatible with GNU nm
LLVM (http://llvm.org/):
  LLVM version 15.0.0git
  Optimized build.
  Default target: x86_64-apple-darwin21.4.0
  Host CPU: znver3
```

**Comment by nikic** [comment]
I don't think this should require an `-opaque-pointers` option. We should be detecting this automatically.

Tracing through the code, I think the relevant method that llvm-nm uses is getBitcodeFileContents, which does some kind of partial read of the bitcode and probably doesn't hit the logic for enabling opaque pointers. Possibly we need to extend that method to look at types or include a dedicated flag in the bitcode.

**Comment by Brandon-T** [comment]
> I don't think this should require an `-opaque-pointers` option. We should be detecting this automatically.
> 
> Tracing through the code, I think the relevant method that llvm-nm uses is getBitcodeFileContents, which does some kind of partial read of the bitcode and probably doesn't hit the logic for enabling opaque pointers. Possibly we need to extend that method to look at types or include a dedicated flag in the bitcode.

Hi. We figured it out. We had to add `--no-llvm-bc       - Disable LLVM bitcode reader` and now everything works :o

**Comment by nikic** [comment]
If you pass `--no-llvm-bc` you're probably going to miss the symbols from the bitcode files though. Is that fine for your use-case?

I think we can probably fix this by unconditionally enabling opaque pointers in llvm-nm, because opaque pointers really shouldn't matter for the things it does (and we can read typed pointers in opaque pointer mode, just not the other way around).
